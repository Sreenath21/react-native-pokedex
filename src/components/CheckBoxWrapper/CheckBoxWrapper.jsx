import React, {useContext, useEffect, useState} from 'react';
import {View, Text} from 'react-native';
import CheckBox from '@react-native-community/checkbox';

import {TypesContext} from '../../context/TypesContext';
import {capitalize} from '../../utils/helplerFunctions';
import styles from './CheckBoxWrapper.style';

const CheckBoxWrapper = ({title, name}) => {
  const [toggleValue, setToggleValue] = useState(false);
  const {filteredTypes, checked, setChecked, genders} =
    useContext(TypesContext);

  useEffect(() => {
    if (title === 'Type') {
      filteredTypes.some(type => type === name) && setToggleValue(true);
    }
    if (title === 'Gender') {
      genders.some(gen => gen === name) && setToggleValue(true);
    }
  }, []);

  const onValueChange = newValue => {
    setToggleValue(newValue);
    if (newValue) {
      setChecked([...checked, name]);
    } else {
      setChecked(checked.filter(type => type !== name));
    }
  };

  return (
    <View style={styles.container}>
      <CheckBox
        value={toggleValue}
        onValueChange={onValueChange}
        boxType="square"
        animationDuration={0.01}
        style={styles.checkbox}
      />

      <Text style={styles.text}>{capitalize(name)}</Text>
    </View>
  );
};

export default CheckBoxWrapper;
