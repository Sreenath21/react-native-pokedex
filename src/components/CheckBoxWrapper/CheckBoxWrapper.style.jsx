import {StyleSheet} from 'react-native';
import colors from '../../utils/colors';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    columnGap: 10,
    // flex: 1,
    width: '40%',
  },
  checkbox: {
    width: 20,
    height: 20,
  },
  text: {
    color: colors.primary,
  },
});
