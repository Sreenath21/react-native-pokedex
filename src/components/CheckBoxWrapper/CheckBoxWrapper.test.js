import React from 'react';
import CheckBoxWrapper from './CheckBoxWrapper';
import {render} from '@testing-library/react-native';
import TypesContextProvider from '../../context/TypesContext';

it('Renders Default Types', () => {
  const screen = render(<CheckBoxWrapper name="Normal" />, {
    wrapper: TypesContextProvider,
  });

  screen.getByText('Normal');
  expect(screen).toMatchSnapshot();
});
