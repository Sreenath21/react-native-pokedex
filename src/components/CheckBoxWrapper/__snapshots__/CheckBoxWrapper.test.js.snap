// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Default Types 1`] = `
<View
  style={
    {
      "columnGap": 10,
      "flexDirection": "row",
      "width": "40%",
    }
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      {
        "checked": false,
        "disabled": false,
      }
    }
    accessibilityValue={
      {
        "text": "off",
      }
    }
    accessible={true}
    pointerEvents="auto"
  >
    <RNCCheckbox
      animationDuration={0.01}
      boxType="square"
      forwardedRef={null}
      onValueChange={[Function]}
      style={
        [
          {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          {
            "height": 20,
            "width": 20,
          },
        ]
      }
      value={false}
    />
  </View>
  <Text
    style={
      {
        "color": "#391363",
      }
    }
  >
    Normal
  </Text>
</View>
`;
